### ISSUE TITLE
Proposal: `norm16-{filterable,resolve}` extensions

### ISSUE THREAD

**Issue by teoxoy** [gpuweb/gpuweb#3839]
Proposal to add 2 extensions that enable filtering and resolve respectively for the following formats:
- `r16unorm`
- `r16snorm`
- `rg16unorm`
- `rg16snorm`
- `rgba16unorm`
- `rgba16snorm`

Note that both extensions depend on [`formats-tier-1`](https://github.com/gpuweb/gpuweb/issues/3837).

Extensions:
- `norm16-filterable`

  Metal: Available
  D3D FL11: Available
  Vulkan: Needs `SAMPLED_IMAGE_FILTER_LINEAR` (loses Samsung Xclipse 920 and a lot of Mali devices + all Adreno 5XX devices due to [`formats-tier-1`](https://github.com/gpuweb/gpuweb/issues/3837))

- `norm16-resolve`

  Metal: Available only on Mac2
  D3D FL11: Available
  Vulkan: Available

---

We don't seem to have precedent for dependant extensions. We can either require that usage of these extensions also requires the user to enable `formats-tier-1` or enable `formats-tier-1` automatically.

**Comment by kainino0x** [comment]
Note: this new feature would be dependent on another new feature from #3001 or #3887.

**Comment by MikhailGorobets** [comment]
Are there any updates regarding the extension? In Google Chrome, up to version 147, UNORM/SNORM formats were not validated against the filterable flag, so I did not even realize that filtering was unavailable for UNORM/SNORM. I use 3D textures for volume path tracing, and switching from hardware trilinear interpolation to manual interpolation reduces performance by 30–40% in my use cases while also making the code more complex.

**Comment by pohlt** [comment]
I would like to chime in and kindly ask for filterable U/SNORM16 textures.
Medical data often comes as (U)INT16. If it's just a single static image with 10^6 pixels, you can convert it once to float and be done, but 10^6 pixels @ 60 Hz or path tracing through a 3D volume (which I also have to do, @MikhailGorobets ) is a different story altogether.

**Comment by MikhailGorobets** [comment]
> I would like to chime in and kindly ask for filterable U/SNORM16 textures.
> Medical data often comes as (U)INT16. If it's just a single static image with 10^6 pixels, you can convert it once to float and be done, but 10^6 pixels @ 60 Hz or path tracing through a 3D volume (which I also have to do, [@MikhailGorobets](https://github.com/MikhailGorobets) ) is a different story altogether.

Medical image data can be stored in different encoded formats. After decoding the pixel data according to the DICOM Transfer Syntax, it is usually available in its stored integer representation, often UINT16 or INT16 depending on the DICOM metadata.

For CT, the stored values can typically be converted to Hounsfield Units using Rescale Slope and Rescale Intercept, and then normalized over a practical range such as [-1024, 3071]. MRI data, on the other hand, usually does not have a universal physical intensity scale, so it is often normalized using percentile-based ranges such as p1 and p99.

The main issue with using Float16 for normalized volume data is that its precision is not uniformly distributed like in fixed-point formats. Float16 provides denser precision near zero and coarser absolute precision at larger magnitudes, whereas a fixed-point format such as UNORM16 provides a uniform step size across the entire normalized range.

**Comment by fhkremerP** [comment]
Agreed, filterable unorm16 is used a lot in medical 3d volume rendering (for example CT scans). The alternative now is converting the native uint16_t data to float, and then uploading it, which doubles the size of the already huge textures.
